nibabel/tests/test_image_api.py

```diff
@@ -175,12 +189,21 @@ def validate_dtype(self, imaker, params):
                 rt_img = bytesio_round_trip(img)
             assert_equal(rt_img.get_data_dtype().type, params['dtype'])
         # Setting to a different dtype
-        img.set_data_dtype(np.float32)  # assumed supported for all formats
-        assert_equal(img.get_data_dtype().type, np.float32)
+        new_dtype = np.float32
+        # some Image types accept only a few datatypes and shapes
+        # so we check and force a type change to a compatible dtype
+        try:
+            supported_dtypes = supported_np_types(img.header_class())
+            if new_dtype not in supported_dtypes:
+                new_dtype = supported_dtypes.pop()
+        except:
```

Comment:
Can you specify the expected errors here?  Bare `except` is a bit frightening...